### ISSUE TITLE
Greylisted messages bypass multimap filters

### ISSUE THREAD

**Issue by edmondburnett** [rspamd/rspamd#4990]
Hello, I'm on rspamd ` 3.8.4`. I've been trying to figure out why some obvious spam messages that should have been flagged are able to completely bypass my multimap filters. And I noticed that the common thing among them is that these are messages that have been caught in the greylisted, or `soft reject` state.

Eventually, the message gets retried a few times and gets the `add header` flag on it, continuing to bypass my custom filters and landing in the user's Inbox.

The expected behaviour is that all incoming messages should be checked with the multimaps, and this should take priority over greylisting.

All of my multimaps are setup as prefilters, and work fine whenever the message is not greylisted:

```
KNOWN_HEADERS {
    type = "content";
    filter = "headers";
    map = "${CONFDIR}/override.d/known_spam_headers.inc";
    prefilter = true;
    score = 20;
    action = "reject";
    regexp = true;
    description = "Blacklisted headers regex";
}

KNOWN_BODY {
    type = "content";
    filter = "body";
    map = "${CONFDIR}/override.d/known_spam_body.inc";
    prefilter = true;
    score = 20;
    action = "reject";
    regexp = true;
    description = "Blacklisted body regex";
}

SCORE_INCREASE {
    description = "Expressions that increase spam score";
    type = "content";
    filter = "body";
    map = "${CONFDIR}/override.d/score_increase.inc";
    regexp = true;
    score = 3;
    multi = true;
    prefilter = true;
}

```

You can see that this example message seems to have had the other checks run on it, but was not checked with any of the multimaps:

![Screenshot 2024-05-27 at 7 56 25 AM](https://github.com/rspamd/rspamd/assets/1209199/a3e4a02f-b36f-4e36-9273-a47738058e43)

The rest of my config is stock. I searched but couldn't locate any similar issue.

Thank you!

**Comment by fatalbanana** [comment]
As things stand today the `prefilter` setting is unused & what the docs call pre-filter maps are not in fact configured as prefilters.

**Comment by edmondburnett** [comment]
Is there some other way to put greylisting later in the module priority? Or do I need to disable the greylist module to get the expected behaviour?

**Comment by edmondburnett** [comment]
Seems to be fixed as of 3.10.0, thank you!
